### PR TITLE
Document Output Reports code

### PR DESCRIPTION
**Pull request by schalkms**
as noted in #2869

### CODE REVIEW

File: detekt-report-txt/src/main/kotlin/io/github/detekt/report/txt/TxtOutputReport.kt [hunk 1]
```diff
@@ -3,6 +3,10 @@ package io.github.detekt.report.txt
 import io.gitlab.arturbosch.detekt.api.Detektion
 import io.gitlab.arturbosch.detekt.api.OutputReport
 
+/**
+ * Contains rule violations in a plain text report similar to a log file.
```
Comment (cortinico) [inline detekt-report-txt/src/main/kotlin/io/github/detekt/report/txt/TxtOutputReport.kt:7]:
```suggestion
 * Contains rule violations and metrics in a plain text report similar to a log file.
```

File: detekt-report-xml/src/main/kotlin/io/github/detekt/report/xml/XmlOutputReport.kt [hunk 2]
```diff
@@ -6,7 +6,8 @@ import io.gitlab.arturbosch.detekt.api.OutputReport
 import io.gitlab.arturbosch.detekt.api.Severity
 
 /**
- * Generates an XML report following the structure of a Checkstyle report.
+ * Contains rule violations in an XML format. The report follows the structure of a Checkstyle report.
```
Comment (cortinico) [inline detekt-report-xml/src/main/kotlin/io/github/detekt/report/xml/XmlOutputReport.kt:9]:
```suggestion
 * Contains rule violations and metrics in an XML format. The report follows the structure of a Checkstyle report.
```

File: detekt-report-txt/src/main/kotlin/io/github/detekt/report/txt/TxtOutputReport.kt (hunk 1 above)
Comment (schalkms) [reply detekt-report-txt/src/main/kotlin/io/github/detekt/report/txt/TxtOutputReport.kt:7]: This report only contains rule violations and no metrics. 😉
That's the reason for documenting the reports now.

File: detekt-report-xml/src/main/kotlin/io/github/detekt/report/xml/XmlOutputReport.kt (hunk 2 above)
Comment (schalkms) [reply detekt-report-xml/src/main/kotlin/io/github/detekt/report/xml/XmlOutputReport.kt:9]: This report only contains rule violations and no metrics. 😉
That's the reason for documenting the reports now.

**Pull request merged**